### PR TITLE
Only mitigate finding if previously active

### PR DESCRIPTION
**Pull request by madchap**
Fixes https://github.com/DefectDojo/django-DefectDojo/issues/3522

### CODE REVIEW

**Comment by valentijnscholten** [comment]
Should the same check be applied in the API (haven't checked what's there now).
Also it looks like finding is set as mitigated when marked as false positive? (existing code)

**Comment by madchap** [comment]
> Should the same check be applied in the API (haven't checked what's there now).
> Also it looks like finding is set as mitigated when marked as false positive? (existing code)

I'll take a look in the API.

Right, when FP, it's currently mitigated :man_shrugging: I didn't want to change it at first, but maybe this can be something to do now.

**Comment by valentijnscholten** [comment]
not sure. it might have some benefits to mark the finding as mitigated when marked as false_p. SLA counter stops etc.
but in that case it should be the same for out_of_scope

**Comment by madchap** [comment]
Tested the API, it seems it never had that behavior actually. It's not trying to assume anything.

**Comment by valentijnscholten** [comment]
There is something around mitigated but it's about keeping the is_Mitigated in sync with mitigated. Maybe we should add the same logic there for false poistive and out of scope.

File: dojo/finding/helper.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        new_state_finding.mitigated = datetime.datetime.now()
+        new_state_finding.mitigated_by = request_user
+        if settings.USE_TZ:
+            new_state_finding.mitigated = timezone.make_aware(new_state_finding.mitigated, timezone.get_default_timezone())
+    elif not new_state_finding.is_Mitigated and new_state_finding.mitigated is not None:
+        finding_status_changed = True
+        new_state_finding.mitigated = None
+        new_state_finding.mitigated_by = None
+
+    logger.info("Finding status after update: {}".format(new_state_finding.status()))
```
Comment (valentijnscholten) [inline dojo/finding/helper.py:45]: lower log level or remove?

**Review by valentijnscholten — CHANGES_REQUESTED** [review]
there are some other places that need to use the new logic: 
image

File: dojo/finding/helper.py (hunk 1 above)
Comment (madchap) [reply dojo/finding/helper.py:45]: It was supposed to be debug definitely.

**Review by valentijnscholten — APPROVED** [review]

**Pull request merged**